### ISSUE TITLE
Allow users to prioritize work

### ISSUE THREAD

**Issue by annawoodard** [Parsl/parsl#108]
In cases where a user is running a set of disjoint workflows, it might be helpful if they could assign priorities such that specified workflows would preferentially be assigned resources. For example, consider the following DAGs:

```
red1 -> red2a, red2b, red2c, red2d -> red3
blue1 -> blue2a, blue2b, blue2c, blue2d -> blue3
green1 -> green2a, green2b, green2c, green2d -> green3
```

Say each task takes ~1 hour and you have enough resources to run four tasks concurrently (toy example here, not meant to be realistic). Consider the following assignments:

**case 1:**
```
hour 1: red1, blue1, green1
hour 2: red2a, red2b, red2c, red2d 
hour 3: red3, blue2a, blue2b, blue2c
hour 4: blue2d, green2a, green2b, green2c
hour 5: blue3, green2d
hour 6: green3
```

**case 2:**
```
hour 1: red1, blue1, green1
hour 2: red2a, red2b, blue2a, green2a
hour 3: blue2b, blue2c, green2b, green2c
hour 4: red2c, red2d, blue2d, green2d
hour 5: red3, blue3, green3
```

In case 1, the red workflow is finished in 3 hours and everything is finished in six hours. In case 2, the red workflow finishes at the same time as everything else, in five hours. For large scale processing, the differences could be much more significant. It could be useful to have one workflow completely finished so the researcher could, for example, use the result to test the next step in their development chain.

As requested by @djf604.

**Comment by annawoodard** [comment]
Hello issue #108, old friend. I am now encountering an urgent need for this in a workflow I'm developing. We are processing a large number of samples of genomic data. Processing each sample produces intermediate files which take up ~10x the final output size. At the end of processing each sample, we delete intermediate files. When a worker becomes available, we need Parsl to favor finishing an existing sample over starting a new sample, so that we can minimize how many samples are pending at any given time (thus the size of the intermediate files).

**Comment by danielskatz** [comment]
I have nothing against this solution, but just want to point out the an alternative solution would be for Parsl to be given hints about sizes of input and output files that it could use to schedule tasks to minimize storage - see https://doi.org/10.1155/2007/701609

**Comment by benclifford** [comment]
I have prototyped this for htex in LSST-branch parsl and been using it with big workflows for a few months.

**Comment by benclifford** [comment]
The htex prioritisation prototype is now available in a PR #1933. I've also tagged workqueue because I think this is something it should be able to do too.

**Comment by benclifford** [comment]
I closed #1933 because of lack of enthusiasm. However, PR #2067 which is merged into master allows priority to be specified with the work queue executor, using WQ's priority mechanism. I think that resolves this issue, at least for tasks big enough to run in WQ.
